funding/batch.go

```diff
@@ -0,0 +1,512 @@
+				Shim: &lnrpc.FundingShim_PsbtShim{
+					PsbtShim: &lnrpc.PsbtShim{
+						PendingChanId: pendingChanID[:],
+						NoPublish:     true,
+					},
+				},
+			},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("error parsing channel %d: %v",
+				idx, err)
+		}
+
+		// Prepare the stuff that we'll need for the internal PSBT
+		// funding.
+		fundingReq.PendingChanID = pendingChanID
+		fundingReq.ChanFunder = chanfunding.NewPsbtAssembler(
+			btcutil.Amount(rpcChannel.LocalFundingAmount), nil,
+			b.cfg.NetParams, false,
+		)
+
+		b.channels = append(b.channels, &batchChannel{
+			pendingChanID: pendingChanID,
+			fundingReq:    fundingReq,
+		})
+	}
+
+	// From this point on we can fail for any of the channels and for any
+	// number of reasons. This deferred function makes sure that the full
+	// operation is actually atomic: We either succeed and publish a
+	// transaction for the full batch or we clean up everything.
+	defer b.cleanup(ctx)
+
+	// Now that we know the user input is sane, we need to kick off the
+	// channel funding negotiation with the peers. Because we specified a
+	// PSBT assembler, we'll get a special response in the channel once the
+	// funding output script is known (which we need to craft the TX).
+	b.wg.Add(len(b.channels))
+	b.negotiateErr = nil
+	for _, channel := range b.channels {
+		channel.updateChan, channel.errChan = b.cfg.ChannelOpener(
+			channel.fundingReq,
+		)
+
+		// Launch a goroutine that waits for the initial response on
+		// either the update or error chan.
+		go b.waitForUpdate(channel, true)
+	}
+
+	// Wait for all goroutines to report back. Any error at this stage means
+	// we need to abort.
+	b.wg.Wait()
+	if b.negotiateErr != nil {
+		return nil, fmt.Errorf("error batch opening channel, initial "+
+			"negotiation failed: %v", b.negotiateErr)
+	}
+
+	// We can now assemble all outputs that we're going to give to the PSBT
+	// funding method of the wallet kit server.
+	txTemplate := &walletrpc.TxTemplate{
+		Outputs: make(map[string]uint64),
+	}
+	for _, channel := range b.channels {
+		txTemplate.Outputs[channel.fundingAddr] = uint64(
+			channel.fundingReq.LocalFundingAmt,
+		)
+	}
+
+	// Great, we've now started the channel negotiation successfully with
+	// all peers. This means we know the channel outputs for all channels
+	// and can craft our PSBT now. We take the fee rate and min conf
+	// settings from the first request as all of them should be equal
+	// anyway.
+	firstReq := b.channels[0].fundingReq
+	feeRateSatPerKVByte := firstReq.FundingFeePerKw.FeePerKVByte()
+	fundPsbtReq := &walletrpc.FundPsbtRequest{
+		Template: &walletrpc.FundPsbtRequest_Raw{
+			Raw: txTemplate,
+		},
+		Fees: &walletrpc.FundPsbtRequest_SatPerVbyte{
+			SatPerVbyte: uint64(feeRateSatPerKVByte) / 1000,
+		},
+		MinConfs:         firstReq.MinConfs,
+		SpendUnconfirmed: firstReq.MinConfs == 0,
+	}
+	fundPsbtResp, err := b.cfg.WalletKitServer.FundPsbt(ctx, fundPsbtReq)
```

Comment:
It can be a privacy concern but also a privacy win, depending on how you use this. Since you already have the same with Pool, using this functionality will make your channel open transactions look like Pool batches which could also be a good thing.

Reply:
Okay, I added a section in the `psbt.md` document.